### PR TITLE
feat: add mosaic brush in menu

### PR DESCRIPTION
**Pull request by jkcs**
### Please check if the PR fulfills these requirements

- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has description for the breaking change

### Description
When I used `tui.iamge-editor`, I found that it did not have the feat of mosaic brush. This feat is very important to us, so I implemented it myself. However, this PR does not contain SVG icons. I hope a partner can provide a set of SVG icons. You can mark it as help wanted when you receive this PR. This PR is just missing the SVG icon。

### CODE REVIEW

**Comment by jkcs** [comment]
image

**Comment by lja1018** [comment]
@jkcs 
I think it's such a wonderful feature. Thank you.
I will proceed with the code review as soon as possible.

Thank you again.

**Comment by jkcs** [comment]
> @jkcs I think it's such a wonderful feature. Thank you. I will proceed with the code review as soon as possible.
> 
> Thank you again.

I think of a question. Is it a bit inappropriate if the setsrc method is directly marked as mosaic.

**Comment by lja1018** [comment]
@jkcs 
I was going to leave that part as a code review.
I hope `setSrc` will be changed to another name.

**Comment by jkcs** [comment]
@lja1018 
I know it's a problem. But I didn't think of a better name.
Would it be better to replace `setSrc` with `setMosaic`?

**Comment by lja1018** [comment]
@jkcs 
Personally, I think `setMosaic` would be okay.

**Comment by jkcs** [comment]
OK, I'll fix it as soon as possible.

File: apps/image-editor/src/js/command/setMosaic.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+   * @param {string} imgUrl - Image Url
+   * @returns {Promise}
+   */
+  execute(graphics, imgUrl, type) {
+    const prevImage = graphics.getCanvasImage();
+    const prevImageWidth = prevImage ? prevImage.width : 0;
+    const prevImageHeight = prevImage ? prevImage.height : 0;
+    const objects = graphics
+      .removeAll(false)
+      .filter((objectItem) => objectItem.type !== 'cropzone');
```
Comment (lja1018) [inline apps/image-editor/src/js/command/setMosaic.js:23]:
```suggestion
      .filter(({ type }) => type !== 'cropzone');
```
how about this?

File: apps/image-editor/src/js/component/mosaicDrawing.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    if (isOut(x, y)) {
+      return;
+    }
+
+    const color = this.getXY(image, x, y);
+
+    // eslint-disable-next-line no-plusplus
+    for (let k = 0; k < this.width; k++) {
+      // eslint-disable-next-line no-plusplus
+      for (let l = 0; l < this.width; l++) {
```
Comment (lja1018) [inline apps/image-editor/src/js/component/mosaicDrawing.js:117]:
```suggestion
    for (let k = 0; k < this.width; k += 1) {
      for (let l = 0; l < this.width; l += 1) {
```
This is our convention, so you have to modify it like this.

File: apps/image-editor/src/js/component/mosaicDrawing.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        if (!isOut(x + l, y + k)) this.setXY(image, x + l, y + k, color);
```
Comment (lja1018) [inline apps/image-editor/src/js/component/mosaicDrawing.js:118]:
```suggestion
        if (!isOut(x + l, y + k)) {
          this.setXY(image, x + l, y + k, color);
        }
```
This is also our convention.

**Review by lja1018 — COMMENTED** [review]
I'm going to finish the review.
I can also add a commit to adding icons next time.

File: apps/image-editor/src/js/command/setMosaic.js (hunk 1 above)
Comment (jkcs) [reply apps/image-editor/src/js/command/setMosaic.js:23]: Blocking rules need to be added
// eslint-disable-next-line no-shadow

Comment (lja1018) [reply apps/image-editor/src/js/command/setMosaic.js:23]: You're right. I didn't see the `type` above. 😅

File: apps/image-editor/src/js/component/mosaicDrawing.js (hunk 3 above)
Comment (jkcs) [reply apps/image-editor/src/js/component/mosaicDrawing.js]: It‘s fixed

**Comment by jkcs** [comment]
@lja1018 by the way, I accidentally saw issues #693 close it

**Comment by lja1018** [comment]
@jkcs 
Thank you for your contribution. This will be merged soon.

**Comment by jkcs** [comment]
I suggest not merging without svg icons

**Comment by lja1018** [comment]
@jkcs 
OK. I won't merge this PR without SVG icon.
Why don't you look it up at the [flaticon](https://www.flaticon.com/)?

**Comment by jkcs** [comment]
I'm sorry. My aesthetic is very poor. I don't know if my choice is appropriate, and I don't have a premium subscription to flation. If I have time, I can try to draw a mosaic icon.

**Comment by lja1018** [comment]
@jkcs 
It's okay. :) I'll look for it, too.

**Comment by jkcs** [comment]
There is an exception in this Feat

**Comment by zhengki** [comment]
Is this function online? How to use it, please

**Pull request closed without merging**